src/Bicep.Decompiler/TemplateConverter.cs

```diff
@@ -468,7 +474,20 @@ private SyntaxBase ParseFunctionExpression(FunctionExpression expression)
                         }
 
                         var stringVal = jTokenExpression.Value.Value<string>()!;
-                        var resolved = nameResolver.TryLookupName(NameType.Parameter, stringVal) ?? throw new ArgumentException($"Unable to find parameter {stringVal}");
+                        var resolved = nameResolver.TryLookupName(NameType.Parameter, stringVal);
+                        if (resolved is null)
+                        {
+                            if (options.AllowMissingParamsAndVars)
+                            {
+                                resolved = nameResolver.TryRequestName(NameType.Parameter, stringVal)
+                                ?? throw new ArgumentException($"Unable to pick unique name for missing parameter {stringVal}");
```

Comment:
Could you please add a test for this scenario?

Reply:
I tried, I'm not sure it's possible to trip that code but have left it in to be defensive.